lib/app.js

```diff
@@ -205,11 +204,85 @@ module.exports = function(tilelive, options) {
             headers["content-encoding"] = headers["content-encoding"] || "gzip";
           }
 
-          res.set(populateHeaders(headers, params, { 200: true }));
-          return res.send(data);
+          return callback(null, data, populateHeaders(headers, params, { 200: true }));
         });
       });
     });
+  };
+
+  app.get(tilePattern, function(req, res, next) {
+    var z = req.params.z | 0,
+        x = req.params.x | 0,
+        y = req.params.y | 0,
+        scale = getScale(req.params.scale),
+        format = req.params.format;
+    return getTile(z, x, y, scale, format, function(err, data, headers) {
+        if (err && !err.status) {
+          return next(err);
+        }
+        res.set(headers);
+        res.status((err && err.status) || 200);
+        return res.send((err && err.message) || data);
+    }, res, next);
+  });
+
+  var processStaticMap = function(areaParams, req, res, next) {
+    var scale = getScale(req.params.scale),
+        format = req.params.format,
+        params = {
+          zoom: req.params.z | 0,
+          scale: scale,
+          bbox: areaParams.bbox,
+          center: areaParams.center,
+          format: format,
+          getTile: function(z, x, y, callback) {
+            return getTile(z, x, y, scale, format, function(err, data, headers) {
+              if (!err && data == null) {
```

Comment:
You're right, but I can't think of any easy way to achieve this. Passing `null` data to abaculus would cause it to fail; not calling the callback would cause waiting until timeout. Only valid option is to either patch this somehow in abaculus or create empty 1x1 transparent image and pass this instead of the 404, but we would probably need to have these in various formats.

The only drawback of this behavior is that we return 404 even if only some of the required tiles are missing. I think this does not block the basic usage and does not need to be fixed in this PR.